Userland/Libraries/LibPDF/Document.cpp

```diff
@@ -109,39 +109,62 @@ PDFErrorOr<Page> Document::get_page(u32 index)
     auto page_object = TRY(get_or_load_value(page_object_index));
     auto raw_page_object = TRY(resolve_to<DictObject>(page_object));
 
-    auto resources = TRY(get_inheritable_object(CommonNames::Resources, raw_page_object))->cast<DictObject>();
-    auto contents = TRY(raw_page_object->get_object(this, CommonNames::Contents));
-
-    auto media_box_array = TRY(get_inheritable_object(CommonNames::MediaBox, raw_page_object))->cast<ArrayObject>();
-    auto media_box = Rectangle {
-        media_box_array->at(0).to_float(),
-        media_box_array->at(1).to_float(),
-        media_box_array->at(2).to_float(),
-        media_box_array->at(3).to_float(),
-    };
-
-    auto crop_box = media_box;
-    if (raw_page_object->contains(CommonNames::CropBox)) {
-        auto crop_box_array = TRY(raw_page_object->get_array(this, CommonNames::CropBox));
+    RefPtr<DictObject> resources;
+    auto maybe_resources_object = TRY(get_inheritable_object(CommonNames::Resources, raw_page_object));
+    if (maybe_resources_object.has_value())
+        resources = maybe_resources_object.value()->cast<DictObject>();
+    else
+        resources = adopt_ref(*new DictObject({}));
+
+    RefPtr<Object> contents;
+    if (raw_page_object->contains(CommonNames::Contents))
+        contents = TRY(raw_page_object->get_object(this, CommonNames::Contents));
+
+    Rectangle media_box;
+    auto maybe_media_box_object = TRY(get_inheritable_object(CommonNames::MediaBox, raw_page_object));
+    if (maybe_media_box_object.has_value()) {
+        auto media_box_array = maybe_media_box_object.value()->cast<ArrayObject>();
+        media_box = Rectangle {
+            media_box_array->at(0).to_float(),
+            media_box_array->at(1).to_float(),
+            media_box_array->at(2).to_float(),
+            media_box_array->at(3).to_float(),
+        };
+    } else {
+        // As most other libraries seem to do, we default to the standard
+        // US letter size of 8.5" x 11" (612 x 792 Postscript units).
+        media_box = Rectangle {
+            0, 0,
+            612, 792
+        };
```

Comment:
(a4 seem to be more european—not really sure which would be better then! 😅)

Reply:
I've previously checked this with MuPDF and whatever Firefox uses, this seems to be the safest guess.